### PR TITLE
Update the entire codebase to ensure adherance to eslint per wp core standards

### LINKED ISSUE #77: JS strings not translated from minified JS when referencing `window.wp.i18n`
### Reproduction Steps

+ Generate a pot file using `wp i18n make-pot`

### Expected Behavior

+ All JS strings using `wp.i18n` module functions should be included in the generated pot file

### Actual Behavior

+ Only strings that import functions from the `@wordpress/i18n` are included in the pot file, any strings that reference the module in the browser's global scope via `const { __ } = wp.18n;` (for example) are not included.
+ I believe this is because of webpack mangling/minifirication. The `wp.i18n.__()` is rewritten during the minification process. The `make-pot` command cannot parse this (understandably) so the strings are not picked up.
+ This isn't a problem given current build steps that generate the pot file from unminified scripts, but the strings are not available on the core plugin's GlotPress instance when the plugin is included as a library in the LifterLMS Core (as a result of the way strings are generated as described in https://github.com/gocodebox/lifterlms/issues/1473)
+ To resolve this we should replace all direct refrences to the module in the browser's global / window scope with imports

### PR DESCRIPTION
**Pull request by thomasplevy**
## Description

Replaces all (maybe most) references to modules on `window.wp` with imports from the equivalent `@wordpress` package.

Fixes #77

## How has this been tested?

Manually

## Types of changes

Bugfix

## Checklist:

- [x] My code has been tested.
- [x] My code passes all existing automated tests. 
- [x] My code follows the LifterLMS Coding & Documentation Standards.

### CODE REVIEW

File: src/js/blocks.js [hunk 1]
```diff
@@ -5,6 +5,7 @@
  * @since 1.5.0 Register blocks conditionally based on post type.
  * @since 1.6.0 Register form field blocks.
  * @since 1.7.0 Refactor into multiple files for clarity and organization.
+ * @since [version] Disable import of incomplete modue `./formats/merge-codes`;
```
Comment (eri-trabiccolo) [inline src/js/blocks.js]:
```suggestion
 * @since [version] Disable import of incomplete modue `./formats/merge-codes`.
```

**Review by eri-trabiccolo — CHANGES_REQUESTED** [review]
Mmm,
what's the minimum wp version we're going to support after this changes?

**Comment by thomasplevy** [comment]
> Mmm,
> what's the minimum wp version we're going to support after this changes?

This question just made my head explode... I had totally forgotten that even though WP throws deprecation warnings... the replacements aren't backported and if we follow the advice of the deprecation warnings we create issues like #62 which are a nightmare to figure out how to fix. Advice from the core is to *include* the missing packages in our build and load it conditionally in the required version of WP.

This isn't terribly complicated to do but the problem then becomes that we need to know what versions of WP a given package was shipped with in order to figure out which packages we need to backport and load conditionally.

Since we're about to drop support for 5.2 (see https://github.com/gocodebox/lifterlms/pull/1551) we can "safely" say 5.3 will be the minimum supported version.

So then we also need to go and figure out all the packages I've updated for consistencies sake without considering backwards compatiblity.

WHAT A NIGHTMARE THIS INSTANTLY BECAME.

If we review the changelogs on the packages they're recorded in the gutenberg plugin's version number, not in the version number of the core version it was included in. Furthermore, we need to look at individual components imported from the various packages, not just the packages themselves.

So there's no quick way to do this unless we actually have full e2e test coverage to let our tests tell us where things are failing.

I think the best bet is to load up 5.3 - 5.7, load the block editor for a course (and a form) and see if it breaks.

If i doesn't break lets say it works, if it doesn't we'll record it and research the error message on it's own to figure out backwards compat.

**Comment by thomasplevy** [comment]
@eri-trabiccolo I'm sorry for my last commit... I had to get the eslint stuff in place before I lost my mind.

**Pull request merged**